crates/ruff_linter/src/rules/flake8_pyi/rules/future_annotations_in_stub.rs

```diff
@@ -37,9 +43,26 @@ pub(crate) fn from_future_import(checker: &mut Checker, target: &StmtImportFrom)
     } = target
     {
         if name == "__future__" && names.iter().any(|alias| &*alias.name == "annotations") {
-            checker
-                .diagnostics
-                .push(Diagnostic::new(FutureAnnotationsInStub, *range));
+            let mut diagnostic = Diagnostic::new(FutureAnnotationsInStub, *range);
+
+            if checker.settings.preview.is_enabled() {
+                let stmt = checker.semantic().current_statement();
+
+                diagnostic.try_set_fix(|| {
+                    let edit = fix::edits::remove_unused_imports(
+                        std::iter::once("annotations"),
+                        stmt,
+                        None,
+                        checker.locator(),
+                        checker.stylist(),
+                        checker.indexer(),
+                    )?;
+
+                    Ok(Fix::unsafe_edit(edit))
```

Comment:
Arguably I think this is a safe edit. The premise of the rule is that `from __future__ import annotations` has literally no effect in stub files, so on that basis it can't be unsafe to remove the redundant import